framework/tlvf/AutoGenerated/include/tlvf/CmduMessageRx.h

```diff
@@ -25,10 +23,12 @@ class CmduMessageRx : public CmduMessage {
     CmduMessageRx();
     CmduMessageRx(CmduMessageRx &original);
     ~CmduMessageRx();
-
-public:
-    std::shared_ptr<cCmduHeader> parse(uint8_t *buff, size_t buff_len, bool swap_needed = true);
+    bool parse(uint8_t *buff, size_t buff_len, bool swap_needed = true, bool parse_tlvs = false);
     CmduMessageRx &operator=(const CmduMessageRx &) = delete;
+
+private:
+    std::shared_ptr<BaseClass> parseNextTlv();
+    std::shared_ptr<BaseClass> parseWscTlv();
```

Comment:
should be a nitpick - but note that this is initial implementation, which will be soon changed:
We will have a parent `CmduParser` class from which the following will derive:
- `TlvParser` which parses 1905 and mAP TLVs
- `WscAttrParser` which parses WSC attributes for WSC TLVs
- `IntelVsParser` which parses Intel Vendor specific messages

So name changes here are irrelevant, since again - they are bound to be changed soon.